### DISCUSSION TITLE
Is tm1Py REST API traffic identifiable by middleware firewall application filters?

### DISCUSSION THREAD

**Discussion by tbeare** [cubewise-code/tm1py#797] [General]
I'm making a TM1Py REST API connection to port 443 for a source system application and it is blocked by middleware firewalls by default as only approved application traffic is allowed on port 443 on this particular set up. Any non-approved/unidentified application traffic will be dropped.

Is it possible to include TM1Py as an identifiable application in the packet header so it can pass through the filter once it's been configured to accept it? Presumably even if data packet is encrypted the header won't be so it'll be scannable by the firewall?

**Comment by rclapp** [comment]
TM1py includes a header called “context” which TM1 uses to identify it. You could use that header to discern where the traffic is from. If that is not an option you can add headers of your choice to a TM1 connection since it is just an abstraction of the HTTP module

**Comment by MariusWirtz** [comment]
Thanks, @rclapp. Yes. TM1py by default has the `session_context` HTTP header.
If you want to add additional HTTP headers you can do it like this. 

``` python
from TM1py import TM1Service

from TM1py.Services.RestService import RestService

RestService.HEADERS["identifier"] = "Hi, I'm TM1py"

with TM1Service(base_url="https://localhost:12297", user="admin", password="apple") as tm1:
    tm1.server.get_server_name()
```

**Reply by tbeare** [reply]
Thanks both, very useful - will give it a go!
